ETL/woocommerce-etl.sh

```diff
@@ -0,0 +1,116 @@
+#!/usr/bin/env bash
+set -euo pipefail
+IFS=$'\n\t'
+
+if [[ ! -x $(type -p jq) ]]; then
+	cat >&2 <<-MISSING_JQ
+		Please install "jq". On Amazon AMI Linux, install it with:
+		
+		    sudo yum install -y jq
+		
+		See https://stedolan.github.io/jq/download/ for information on other
+		platforms.
+	MISSING_JQ
+	exit 3
+fi
+
+if [[ $(jq --version) == jq-1.[0-4] ]]; then
+	echo >&2 "Please update jq to version 1.5 or higher"
+	exit 4
+fi
+
+usage() {
+	cat >&2 <<-USAGE
+		Usage: $0 [-?|-h|--help] <wordpress_base_url> <basic_auth_username> <basic_auth_password>
+		
+		-? | -h | --help                      Prints this message
+		-a | --all                            Process all orders (overrides total)
+		-p N | --per-page N                   Process orders in batches of N (defaults to 100)
+		-q | --quiet                          Do not print progress information
+		-t N | --total N                      Only include the first N orders (defaults to 1000)
+		-E NAME | --endpoint NAME             WooCommerce endpoint name (defaults to "orders")
+		-A START_DATE | --after START_DATE    Only include orders after this date (E.g. "2017-03-22")
+		-B END_DATE | --before END_DATE       Only include orders before this date (E.g. "2018-04-14")
+		
+	USAGE
+	exit 1
+}
+
+while [[ $# -gt 0 && $1 == -* ]]; do
+	case "$1" in
+	-h | --help) usage ;;
+	-a | --all) ALL=1 ;;
+	-p | --per-page)
+		shift
+		PER_PAGE="${1}"
+		;;
+	-t | --total)
+		shift
+		TOTAL="${1}"
+		;;
+	-E | --endpoint)
+		shift
+		ENDPOINT="${1}"
+		;;
+	-A | --after)
+		shift
+		AFTER="${1}"
+		;;
+	-B | --before)
+		shift
+		BEFORE="${1}"
+		;;
+	-q | --quiet)
+		QUIET=1
+		;;
+	*)
+		echo >&2 "Invalid parameter '$1'"$'\n'
+		usage
+		;;
+	esac
+	shift
+done
+
+WORDPRESS_BASE_URL="${1:-https://example.com}"
+BASIC_AUTH_USER="${2:-example_username}"
+BASIC_AUTH_PASSWORD="${2:-example_password}"
```

Comment:
These are mandatory, right? In that case, there's two things that can be done. First, check for number of parameters:

```bash
if [[ $# -ne 3 ]]; then
    echo >&2 "Missing arguments: expected 3, got $#"
    usage
fi
```

The other is to validate instead of providing a default:

```bash
WORDPRESS_BASE_URL="${1:?Please provide url, auth user and auth password}"
BASIC_AUTH_USER="${2:?Please provide auth user and password}"
BASIC_AUTH_PASSWORD="${2:?Please provide auth password}"
```